apps/src/javalab/JavabuilderConnection.js

```diff
@@ -147,6 +144,12 @@ export default class JavabuilderConnection {
   }
 
   onOpen() {
+    // We need to send an initial message to Javabuilder here in case there was an issue
+    // with our token. Javabuilder will send back an error message with details, but can
+    // only do so once we've sent a message. This is a bit of a hack, but this should only
+    // happen if our token was somehow valid for the initial Javabuilder HTTP request and
+    // then became invalid when establishing the WebSocket connection.
+    this.sendMessage('connected');
```

Comment:
for a console i/o program will this 'connected' message get interpreted as input from the user's program?

Reply:
Oh good call...in theory it shouldn't because it's sent immediately (before the build and run lambda executes)? But maybe we should have special logic in Javabuilder to ignore this specific message. All normal input messages should technically be sent as JSON with an input message type, so I think it should be safe for us to ignore a message with just this text. I'll update this and the Javabuilder PRs.